scripts/karma.coffee

```diff
@@ -48,20 +47,59 @@ class Karma
       delete @cache.things[thing]
     @robot.brain.data.karma = @cache
 
+  cleanKarmaHistory: (actor) ->
+    if @khistory[actor.id]?
+      DAY = 1000 * 60 * 60  * 24
+      today = new Date()
+      @khistory[actor.id] = (x for x in @khistory[actor.id] when (Math.round((today.getTime() - x) / DAY)) <= 7 )
+
+  getKarmaHistory: (actor) ->
+    if @khistory[actor.id]?
+      DAY = 1000 * 60 * 60  * 24
+      today = new Date()
+      todayKarma = (x for x in @khistory[actor.id] when (Math.round((today.getTime() - x) / DAY)) == 0 )
+      yesterdayKarma = (x for x in @khistory[actor.id] when (Math.round((today.getTime() - x) / DAY)) == 1 )
+      return @khistory[actor.id].length + (yesterdayKarma.length*2) + (todayKarma.length*3)
+    else 
+      return 0
+
+  getKarmaHistoryList: (actor) ->
+    if @khistory[actor.id]?
+      return @khistory[actor.id].join()
+    else 
+      return 'none'
+
+  addKarmaHistory: (actor) ->
+    if not @khistory[actor.id]?
+      @khistory[actor.id] = []
+    today = new Date  
+    @khistory[actor.id].push today.getTime()
+    @robot.brain.data.karmaHistory = @khistory
+      
+  getKarmaPower: (actor) ->
+    maxKarmaPower = 3
+    minKarmaPower = 0.01
+    c1 = -0.5
+    c2 = 3
+    c3 = 0.008
+    kv = @getKarmaHistory(actor)
+    return Math.max(minKarmaPower,Math.min(maxKarmaPower,(c1 * Math.log(kv)) + c2 - (kv*c3)))
```

Comment:
Math.log(0) throws an undefined error if the user does not have any @khistory[] entries. log(1) will return 0.